### PR TITLE
Use a newer CXX in CI

### LINKED ISSUE #2: Add CI
Now that this is a fork of snappy-go, we should make sure the tests don't rot.

### PR DESCRIPTION
**Pull request by tamird**
Closes #2.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+dependencies:
+  pre:
+    - ./install-clang-3.6.sh
+
+machine:
+  environment:
+    CXX: /usr/lib/llvm-3.6/bin/clang++
```
Comment (petermattis) [inline circle.yml:8]: Should you set CC as well?

**Comment by petermattis** [comment]
LGTM

File: circle.yml (hunk 1 above)
Comment (tamird) [inline circle.yml:8]: Yep did, thanks

**Comment by bdarnell** [comment]
Why did we need to do this here but not in the main repo?

**Comment by tbg** [comment]
in the main repo everything is inside Docker. This runs on the bare CircleCI vms.

**Comment by bdarnell** [comment]
Ah, so the golang docker base images are newer than the ubuntu 12.04 that circleci uses? 12.04 is still pretty common so it's unfortunate if we can't build there out of the box. What do we need from clang 3.6? Some c++11 feature I assume?

The docker image appears to use gcc 4.9; was there a reason to switch from gcc to clang?

**Comment by tbg** [comment]
> Ah, so the golang docker base images are newer than the ubuntu 12.04 that circleci uses?

Pretty much the first thing we do is install some more stuff on top of the minimal jessie image:

```
16:19 $ cat build/devbase/Dockerfile
FROM golang:1.4.2

MAINTAINER Tobias Schottdorf <tobias.schottdorf@gmail.com>

# Setup the toolchain.
RUN apt-get update -y && apt-get dist-upgrade -y && \
 apt-get install --no-install-recommends --auto-remove -y git build-essential pkg-config file &&\
 apt-get clean autoclean && apt-get autoremove -y && rm -rf /tmp/* /var/lib/{apt,dpkg,cache,log}
```

**Comment by bdarnell** [comment]
Yeah, but in the docker image we're going from no compiler at all to the compiler that comes with debian jessie (no additional apt repositories). Here we're installing a newer compiler than is available in 12.04. This would be a problem for anyone trying to build and deploy on a 12.04 non-dockerized cluster.

**Pull request merged**